internal/config/types.go

```diff
@@ -21,8 +21,7 @@ type ControlServer struct {
 
 // Metrics config.
 type Metrics struct {
-	// ClientType metrics client type e.g. prometheus, datadog.
-	ClientType string `yaml:"client_type" json:"client_type" env:"CLIENT_TYPE" env-default:"noop"`
```

Comment:
What's the plan for rolling this out, as this would remove the Datadog capabilities?

Reply:
I can revert Harry's changes where it defaults to prometheus client (and default to DD), which would look like this:
```
	err := metrics.InitMetricsClient(logger.With(zap.String("component", "metrics-collector")), "datadog")
	if err != nil {
		return fmt.Errorf("init metrics client failure: %w", err)
	}

	defer metrics.Close()
	if config.Metrics.Enabled {
		metricsLogger := logger.With(zap.String("component", "metrics"))
		metricsHandler, err := metrics.CreateHandler(metricsLogger)
		if err != nil {
			return fmt.Errorf("create metrics handler failure: %w", err)
		}
		s, err := server.NewHTTP(server.HTTPOpts{
			Address: ":9090",
			Logger:  metricsLogger,
			Handler: serverUtil.HandlerWithRecovery(metricsHandler, metricsLogger),
		})
		if err != nil {
			return err
		}
		g.AddWithCtxE(s.Run)
	}
```
Or, I can do whatever work is necessary to ingest the prometheus emitted metrics into datadog.

Reply:
[9264fa9](https://github.com/Kong/koko/pull/364/commits/9264fa95b62aa21fc12bae40048520bb79d4ecc5)

Reply:
How are we expecting this to be used if there's no way to configure it to use the newer metrics implementation? We should be feature-flagging this off.

Reply:
@tjasko [e51578f](https://github.com/Kong/koko/pull/364/commits/e51578f5b513005597bc917b3ac20de8076c9f22)

Reply:
Validated server is working locally 
```
/GET localhost:9090/metrics

# HELP go_gc_duration_seconds A summary of the pause duration of garbage collection cycles.
# TYPE go_gc_duration_seconds summary
go_gc_duration_seconds{quantile="0"} 8.416e-06
go_gc_duration_seconds{quantile="0.25"} 3.9667e-05
go_gc_duration_seconds{quantile="0.5"} 5.9041e-05
go_gc_duration_seconds{quantile="0.75"} 9.8332e-05
go_gc_duration_seconds{quantile="1"} 0.000186959
go_gc_duration_seconds_sum 0.000561706
go_gc_duration_seconds_count 8
...
```